tawazi/node/node.py

```diff
@@ -366,13 +370,9 @@ def __call__(self, *args: P.args, **kwargs: P.kwargs) -> RVXN:
         values["tag"] = kwargs.get(ARG_NAME_TAG) or self.tag
         values["unpack_to"] = kwargs.get(ARG_NAME_UNPACK_TO) or self.unpack_to
 
-        new_lxn: LazyExecNode[P, RVXN] = LazyExecNode(**values)
-
-        new_lxn._validate_dependencies()
-
-        exec_nodes[new_lxn.id] = new_lxn
-
-        return new_lxn._usage_exec_node  # type: ignore[return-value]
+        lxn: LazyExecNode[P, RVXN] = LazyExecNode(**values)
+        lxn._validate_dependencies()
```

Comment:
same question than before here, why isn't it done in the init ?

Reply:
because we initailize LazyExecNodes in DAG.config_from_dict. And during that phase, we don't have any data in the global variable exec_nodes

Reply:
hmm then maybe we should consider not having this a method

Reply:
ok I can remove the code from this method and put it directly in the `__call__`...
I only did it to refactor the code a little bit and make the __call__ tinier

Reply:
actually I just further thought about it... I think I have a solution for both cases!